pkgs/top-level/config.nix

```diff
@@ -123,6 +123,32 @@ let
       '';
     };
 
+    ignoreWarningsPackages = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = ''
+        List of packages for which to ignore warnings.
+
+        This value will be overwritten by `ignoreWarningsPredicate`
+
+        See <link xlink:href="https://nixos.org/manual/nixpkgs/stable/#sec-ignore-warnings">Installing packages with warnings</link> in the NixOS manual.
+      '';
+    };
+
+    ignoreWarningsPredicate = mkOption {
+      type = types.functionTo types.bool;
+      default = null;
+      # getEnv part is in check-meta.nix
+      defaultText = literalExpression ''_: false || builtins.getEnv "NIXPKGS_IGNORE_WARNINGS" == "1"'';
```

Comment:
By default it uses `ignoreWarningsPackages`.
It will be easy to keep in sync when you've moved the default implementation here.

Reply:
Hm, the part with `builtins.getEnv` does not make me happy (independently of the usual purity concerns). Because I want to document that this is part of the function, but if I actually make it part of the function, one could accidentally override this behavior away, which is not desirable.